array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,24 @@
+/* exported compact */
+
+
+// dec func 
+// dec var array contains[] containing elements to omit: false, null, NaN, 0, -0, '',  
+// dec var array newArray[] to hold non-omited elements
+// Enters loop for arg array
+// Enters loop for var array, compareing array[i] to each item in contains[]
+//   Pushes non-omited elements new newArray[]
+// returns newArray[]
+
+//This ^ was all wrong HAHA
```

Comment:
You should revise it now that you have the right path! Also, you still have that `contains` variable even though it is not used.